### PR TITLE
Add support for CALayer layout

### PR DESCRIPTION
**Pull request by antoinelamy**
Since every UIView has a backing layer and it's that layer's responsibility alone to manage coordinates and drawing, I added Layoutable conformance to CALayer and delegated UIView's getRect/setRect to it's backing layer. NSView is a slightly different animal and it's not all NSView that has a backing layer so we leave it's current layout system unchanged.

### CODE REVIEW

File: Tests/iOS/PinSafeAreaTests.swift [hunk 1]
```diff
@@ -415,8 +408,8 @@ class PinSafeAreaMoreTestsSpec: QuickSpec {
                 setupWindow(with: navigationController)
 
                 // MATCH safeAreaInsets!
-                expect(mainView.safeAreaInsetsDidChangeCalledCount).to(equal(expectedSafeAreaInsetsDidChangeCalledCount))
-                expect(mainView.subView.safeAreaInsetsDidChangeCalledCount).to(equal(expectedSubViewSafeAreaInsetsDidChangeCalledCount))
+                expect(mainView.safeAreaInsetsDidChangeCalledCount) > 0
```
Comment (lucdion) [inline Tests/iOS/PinSafeAreaTests.swift]: We could also update all other similar cases in safeArea unit tests.

File: Tests/Common/CALayerSpec.swift [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        //
+        // CALayer is already heavily tested since UIView delegate it's layout to it's layer.
+        // Validate only the direct usability.
+        //
+        describe("test CALayer interface") {
+            it("basic pinlayout calls") {
+                aLayer.pin.top(10).left(10).width(20%).height(80%).layout()
+                bLayer.pin.size(CGSize(width: 20, height: 20)).right(of: aLayer, aligned: .center).layout()
+                expect(aLayer.frame).to(equal(CGRect(x: 10, y: 10, width: 80, height: 320)))
+            }
```
Comment (lucdion) [inline Tests/Common/CALayerSpec.swift]: maybe add unit tests that:
* use `pinFrame` 
* use `edge` property
* use `anchor` property

File: Tests/Common/CALayerSpec.swift [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            Pin.logMissingLayoutCalls = false
+        }
+
+        //
+        // CALayer is already heavily tested since UIView delegate it's layout to it's layer.
+        // Validate only the direct usability.
+        //
+        describe("test CALayer interface") {
+            it("basic pinlayout calls") {
+                aLayer.pin.top(10).left(10).width(20%).height(80%).layout()
```
Comment (lucdion) [inline Tests/Common/CALayerSpec.swift]: Is it required to call `layout()`?

File: Tests/iOS/PinSafeAreaTests.swift (hunk 1 above)
Comment (antoinelamy) [reply Tests/iOS/PinSafeAreaTests.swift]: Will do :)

File: Tests/Common/CALayerSpec.swift (hunk 2 above)
Comment (antoinelamy) [reply Tests/Common/CALayerSpec.swift]: Will also do :)

File: Tests/Common/CALayerSpec.swift (hunk 3 above)
Comment (antoinelamy) [reply Tests/Common/CALayerSpec.swift]: Not sure, I took example on the `ObjectiveCSpec` that also manually call `layout()` but it is most likely not needed.

File: README.md [hunk 4]
```diff
@@ -1543,8 +1546,10 @@ PinLayout **support of macOS is not complete**, see here the particularities of
 
 * These methods are currently not supported on macOS, but they will be implemented soon:
 
-	* [`sizeToFit(:FitType)`](#sizeToFit) (Coming soon)
+	* [`sizeToFit(:FitType)`](#sizeToFit) on any view that is not a subclass of NSControl
 	* [`aspectRatio()`](#aspect_ratio) with no parameters (Coming soon)
+    
+* Support for [`sizeToFit(:FitType)`](#sizeToFit) can be added to your custom NSView subclasses. Just make those views conform to the `SizeCalculable` protocol and implement the two required functions.
```
Comment (lucdion) [inline README.md]: Nice 👍

**Pull request merged**